### PR TITLE
1536141: Send events as objects rather than arrays

### PR DESCRIPTION
**Pull request by travis79**
This is currently pointing towards an unmerged schema update on the pipeline side.  Once that [PR](https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/305) lands, `build.gradle` in service-glean will need updated to point the schema hash to the correct master commit.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt [hunk 1]
```diff
@@ -261,34 +260,28 @@ internal object EventsStorageEngine : StorageEngine {
     /**
      * Serializes an event to its JSON representation.
      *
-     * @param category Event category
-     * @param name Event name
-     * @param msSinceStart The ms since the process started
-     * @param extra A `Map<String, String>` of extra values
-     * @return JSONArray representing the event
+     * @param event Event data to serialize
```
Comment (mdboom) [inline components/service/glean/src/main/java/mozilla/components/service/glean/storages/EventsStorageEngine.kt]: Ah, thanks for catching my out-of-date comment!

**Comment by mdboom** [comment]
image

File: components/service/glean/build.gradle [hunk 2]
```diff
@@ -14,7 +14,7 @@ apply plugin: 'kotlin-android'
  * created during unit testing.
  * This uses a specific version of the schema identified by a git commit hash.
  */
-String GLEAN_PING_SCHEMA_GIT_HASH = "3bf541d"
+String GLEAN_PING_SCHEMA_GIT_HASH = "2d5c56f3d95f34c2681babb38e6b06d0bdfed087" // TODO update this back to master git hash when https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/305 lands
```
Comment (Dexterp37) [inline components/service/glean/build.gradle]: Let's make sure to NOT merge this until schemas are updated and this gets changed.

Comment (travis79) [reply components/service/glean/build.gradle]: Of course.  I did update it just now to point to an update in the schema PR that is still open just to verify it still passed all tests with the addition to that repo.  

Once the schema lands I'll get the correct git hash in here and land this one.

**Comment by travis79** [comment]
> image

Yeah, but it was all in comments ;)

**Pull request merged**